src/component-library/NumberInput/NumberInput.tsx

```diff
@@ -1,39 +1,58 @@
 import * as React from 'react';
+
 import { BaseNumberInput } from './NumberInput.style';
-import { InputProps } from 'component-library/Input';
+import { useNumberField } from '@react-aria/numberfield';
+import { useNumberFieldState } from '@react-stately/numberfield';
+
+type Props = { value?: number; defaultValue?: number; step?: number; label?: string; description?: string };
+
+const defaultProps = {};
 
-// `onWheel` prop can't be used with `preventDefault` because
-// React implements passive event listeners.
-const disableChangeOnWheel = (event: MouseEvent) => {
-  event.preventDefault();
-};
+type NativeAttrs = Omit<React.InputHTMLAttributes<Element>, keyof Props>;
 
-type NumberInputProps = InputProps;
+type NumberInputProps = Props & NativeAttrs & Partial<typeof defaultProps>;
 
 const NumberInput = React.forwardRef<HTMLInputElement, NumberInputProps>(
   (props, ref): JSX.Element => {
-    const inputParent = React.useRef<HTMLDivElement | null>(null);
+    const { disabled, readOnly, required, onChange, ...rest } = props;
+
+    const ariaProps = {
+      isDisabled: disabled,
+      isReadOnly: readOnly,
+      isRequired: required,
+      ...rest
+    };
+
+    const state = useNumberFieldState({ ...ariaProps, locale: 'en-US' });
+    const inputRef = React.useRef<HTMLInputElement>(null);
+
+    React.useImperativeHandle(ref, () => inputRef.current as any);
+
+    const { inputProps } = useNumberField(ariaProps, state, inputRef);
 
     React.useEffect(() => {
-      if (!inputParent || !inputParent.current) return;
+      const handleWheel = (e: WheelEvent) => e.preventDefault();
 
-      const currentInputParent = inputParent.current;
+      const input = inputRef.current;
 
-      currentInputParent.addEventListener('wheel', disableChangeOnWheel, { passive: false });
+      input?.addEventListener('wheel', handleWheel, { passive: false });
 
       return () => {
-        currentInputParent.removeEventListener('wheel', disableChangeOnWheel);
+        input?.removeEventListener('wheel', handleWheel);
       };
     }, []);
 
-    return (
-      <div ref={inputParent}>
-        <BaseNumberInput ref={ref} {...props} />
-      </div>
-    );
+    const handleChange: React.ChangeEventHandler<HTMLInputElement> = (e) => {
+      inputProps.onChange?.(e);
```

Comment:
Why us handleChange firing off two onChange events?

Reply:
There is an ìnputProps,onChange` to be used from the `react-stately`. This one is important because it will affect the state inside the `react-stately` hook. The other `onChange` is from props, which is going to be used by `react-hook-forms`

Reply:
Ok, that makes sense - as with the question around the refs above, we would want to handle this in a way to make it clear what's going on as it's not immediately obvious from the code.

Reply:
Possibly, but I'd like this to be really clear, so would be nice if it could be moved out to a utility (as withe the refs) and then commented appropriately (i.e. pretty much copy and paste your first reply into the code!). If that's not possible, then we can add a comment every time it's used. 

Whatever you think though - main thing is that it's clear from the code why we're firing off two change events :)